### ISSUE TITLE
Fractional left or top borders cause large, proportional transparent borders

### ISSUE THREAD

**Issue by SimplGy** [facebook/react-native#14106]
### Description

If a left or top border is `0.5`, all of the borders grow at partial opacity in a way that is not obviously deterministic (maybe proportional?).

### Reproduction Steps and Sample Code

#12401 has several consumers showing reproduction cases. #13305 has another example. Both these issues were closed but this is demonstrably real and reproducible today.

I have isolated cases on expo here:

https://repl.it/ILZ9/26
https://repl.it/ILZ9/31

![react-native-border-bug](https://cloud.githubusercontent.com/assets/548809/26321564/1b9688b2-3edc-11e7-8ad2-7d3ff9474d2b.png)

### Solution

Something is doing strange things with fractional borders. Needs root cause analysis, costing/tradeoff analysis,  solution.

### Additional Information

* React Native version: 46, and also the version in `repl.it` (I can't find where that version is listed)
* Platform: iOS
* Development Operating System: OSX
* Dev tools: Expo, InjelliJ, Simulator, physical device.

(T22099662)

**Comment by SimplGy** [comment]
(moving details/updates to this issue)

These workarounds fix the strange border behavior:

    borderRadius: 1 // any non-fractional border radius
    borderStyle: 'dotted' // or dashed. probably anything except 'solid'

**Comment by shergin** [comment]
@SimplGy Yeah, it should be somewhere inside `RCTBorderDrawing`. May be you even have a proper fix? 😄

**Comment by SimplGy** [comment]
I'm thinking it'll be about rounding any fractional units to something evenly divisible by hairlineWidth (0.33333, 0.5, 1). Or rounding to an Int, if we're in pixel rather than dip space. Haven't found the right place yet.

**Comment by MoOx** [comment]
I am having similar issues and I only able to reproduce on certain devices (ios 6 plus, xperia Z5), so it's also probably related to screen resolution/dpi.

Here is a snippet:

```js
import React, { Component } from 'react';
import { Text, View, StyleSheet } from 'react-native';
import { Constants } from 'expo';

export default class App extends Component {
  render() {
    return (
      <View style={styles.row} />
    );
  }
}

const styles = StyleSheet.create({
  row: {
    position: "absolute",
    top: 100,
    left: 5,
    width: 5,
    height: 200,
    borderTopLeftRadius: 4,
    borderTopRightRadius: 4,
    // here when the border radius are not 0 (not the same as the top)
    // a really weird issue is happening
    // setting toplfet/right to 2 hide the issue
    // so I guess indeed there is some rounding issue that can depend on resolution/dpi
    borderBottomLeftRadius: 0,
    borderBottomRightRadius: 0,
    backgroundColor: "green",
  },
});
    
```

Here is the result on

iphone 6s

![iphone6s](https://user-images.githubusercontent.com/157534/28020426-aaf121c4-6584-11e7-9482-5a9823cbeb60.PNG)

iphone 6s Plus

![iphone6splus](https://user-images.githubusercontent.com/157534/28020430-b393e4f6-6584-11e7-82b2-afca4c1ac84d.PNG)

Here is a playground https://repl.it/ILZ9/46

**Comment by hramos** [comment]
Hi there! This issue is being closed because it has been inactive for a while. Maybe the issue has been fixed in a recent release, or perhaps it is not affecting a lot of people. Either way, we're automatically closing issues after a period of inactivity. Please do not take it personally!

If you think this issue should definitely remain open, please let us know. The following information is helpful when it comes to determining if the issue should be re-opened:

- Does the issue still reproduce on the latest release candidate? Post a comment with the version you tested.
- If so, is there any information missing from the bug report? Post a comment with all the information required by the [issue template](https://github.com/facebook/react-native/blob/master/.github/ISSUE_TEMPLATE.md).
- Is there a pull request that addresses this issue? Post a comment with the PR number so we can follow up.

If you would like to work on a patch to fix the issue, *contributions are very welcome*! Read through the [contribution guide](http://facebook.github.io/react-native/docs/contributing.html), and feel free to hop into [#react-native](https://discordapp.com/invite/0ZcbPKXt5bZjGY5n) if you need help planning your contribution.

**Comment by shergin** [comment]
We still have to fix that.
Promoted to T22099662. 😄

**Comment by SimplGy** [comment]
I understand an icebox policy, but this is easily reproable with multiple sandboxes, seems like a miss to close it.

**Comment by Nikita2k** [comment]
Should work now: https://github.com/facebook/react-native/commit/15179f1798b277c1836441fcf7f3b7f0bd5a4636

**Comment by shergin** [comment]
Fixed in 15179f1798b277c1836441fcf7f3b7f0bd5a4636. 🎉
Huge thanks to @Nikita2k!

**Comment by SimplGy** [comment]
Awesome, thanks guys!
